### PR TITLE
Beyond Con

### PR DESCRIPTION
**Pull request by dadasign**
This is Marcin's and my work for today's Beyond Con.
We were working on 2 new features for our CLI

**beyond query**
It is a command which allows you to easily query our API objects. The simplest example could be: `beyond query users -a 2524 -c` and you'll get a table with a list of users. Another very common use would be `beyond query account_prefs -a 2524 -c` to check the account preferences. I think that this should may it a lot easier to check our api objects without copying tokens to curl or postman. What is more the command supports filtering with --find, sorting (--sort), as well as --limit --page.
Example: `beyond query account_prefs -k .creds -c --find id=6`
You can also export the results to a csv file using `--export` and passing a filename. Check beyond query --help for all the usage details.
If this idea picks it would be interesting to try to support PUT or POST calls, but for now it is for viewing only.

**beyond imp**
This command allows you to easily impersonate an account owner or a user by user email. For example: `beyond imp -k .creds -u someuser@peachworks.com` will get the impersonation link for the user with the given email or `beyond imp -k .creds -a 2524` will impersonate the owner of 2524. You can also pass an '-o' parameter which will instantly open the browser with the impersonation url.
The other addition is that you can use an `-i / --imp` parameter in `dev`,`serve`,`proxy` or `report` commands. If you add this parameter with no value, you will run the command impersonating the owner of the account specified with `-a`. If you pass a value to `--imp`, it should be an email of the user to impersonate. Example: `beyond serve --imp -a 2524 -c -e prod` to serve as the owner of 2524 or `beyond serve --imp someuser@peachworks.com -a 2524 -c -e prod` to run the same account but as a different user.

### CODE REVIEW

**Review by Fellach — APPROVED** [review]

File: bin/beyond [hunk 1]
```diff
@@ -28,15 +28,17 @@ versionCheckPromise
       .command('build', 'Rebuild the project for distribution.')
       .command('clean', 'Remove all distribution files.')
       .command('dev', 'Startup a development session.')
+      .command('imp', 'Summon an imp')
```
Comment (mbebas) [inline bin/beyond]: I get the joke, but we will probably need to update this to something more informative :)

**Review by mbebas — APPROVED** [review]

File: bin/beyond (hunk 1 above)
Comment (Fellach) [reply bin/beyond]: I've suggested to use a simple `Deprecated` as we agreed to hide a bit this option. After all it's an impersonation

**Comment by Fellach** [comment]
There are 2 things "todo".

1. As Kuba mentioned, we thought having POST, PUT and DELETE would be a nice addition, but we've got an issue with tokens. Do you think we can use app token with app_id taken from `.peachrc`? Or maybe we should provide an extra parameter for that. Any thoughts?

2. We could extend the query command and take into account `inculdes` and `findIncludes` but I'm not sure how to present the result using cli-table or CSV?

@darushka, @mbebas, @guyjstitt, @legendary-mich 
Do you have ideas how to easily extend the commands we built right now with @dadasign?

File: bin/beyond (hunk 1 above)
Comment (legendary-mich) [reply bin/beyond]: Security by obscurity is rarely a good idea though. If someone is not an admin they will not be able to impersonate anyway, so I would leave it visible.

Comment (dadasign) [reply bin/beyond]: Oh... I though no one reads this ;-)

**Comment by legendary-mich** [comment]
@Fellach I don't think I'm gonna be using the `query` command for anything but a quick lookup. To make the query option attractive to me it would have to be way simpler to use than curl, and give me a guarantee that it is bug free. If I make a call to the api and it fails I need to be sure that it is because of me, and not because of a bug in the cli.

Also I've created quite a few collections of requests in Postman, and migrating all of them to the cli tool is out of the question to me.

**Pull request merged**